newbank/server/NewBank.java

```diff
@@ -49,13 +49,20 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
 	// commands from the NewBank customer are processed in this method
 	public synchronized String processRequest(CustomerID customer, String request) {
 		if(customers.containsKey(customer.getKey())) {
-			switch(request) {
-			case "SHOWMYACCOUNTS" : return showMyAccounts(customer);
-			case "WITHDRAW" : return withdrawMoney(customer);
-			case "DEPOSIT" : return depositMoney(customer);
-			case "TRANSFERMONEYTOPERSONAL" : return transferMoneyToPersonal(customer);
-			case "TRANSFERMONEYTOEXTERNALACCOUNT" : return transferMoneyToExternal(customer);
-			default : return "FAIL";
+			if (request.equals("SHOWMYACCOUNTS")){
+				return showMyAccounts(customer);
+			}
+			else if (request.startsWith("WITHDRAW")){
+				return withdrawMoney(customer);
+			}
+			else if(request.startsWith("DEPOSIT")){
+				return depositMoney(customer, request);
+			}
+			else if(request.startsWith("TRANSFER")){
+				return transferMoneyToPersonal(customer);
+			}
+			else if (request.startsWith("PAY")){
+				return payThirdParty(customer,request);
```

Comment:
I kept the if statements so as not to have to ask for input for amount, accounts, etc., but we can modify it if you think switch is better

Reply:
Yes I think a switch would be better as it deals with more cases efficiently.